xml/abstract_commands.xml

```diff
@@ -129,9 +129,12 @@
         \item If the hart is halted, the command  sets \FdmAbstractcsCmderr to ``halt/resume'' and does not continue.
         \item Halt the hart. If the hart halts for some other reason (e.g. breakpoint), the command
             sets \FdmAbstractcsCmderr to ``halt/resume'' and does not continue.
-        \item Execute the Program Buffer. If an exception occurs, \FdmAbstractcsCmderr is set to ``exception''
-            and the program buffer execution ends, but the quick access command continues.
-        \item Resume the hart.
+        \item Execute the Program Buffer.
+            If an exception occurs, \FdmAbstractcsCmderr is set to
+            ``exception,'' the Program Buffer execution ends, and the hart is
+            halted with \FcsrDcsrCause set to 3.
```

Comment:
dcsr.cause=3 is "The debugger requested entry to Debug Mode using haltreq."  That doesn't seem to accurately reflect what happened.  On the other hand, there's not a lot of free encoding space for a new cause.  What are your thoughts?

Reply:
Yes, nothing really fits. I just realized I should mention that it also sets cmderr and then there is no ambiguity why the halt occurred.

Reply:
Oh, it already says that. And right there, no less.

I think this isn't ideal, but I don't want to use another dcsr.cause encoding (we have 2 left) for such a rare case.

Reply:
I agree that it's not ideal but it's probably fine.  The debugger ought to know whether it has been doing quick accesses and/or haltreq.  If the hart is halted with cause=3 and cmderr=1 then either:
- it made a haltreq and something else went wrong before or after the halt
- it did a quick access and got an exception

If the debugger wants to do a quick access and then a haltreq, I think it can disambiguate the two scenarios as long as it reads cmderr in between.